tensorflow/core/kernels/dml_lrn_op.cc

```diff
@@ -128,4 +130,107 @@ TF_CALL_float(REGISTER_KERNEL);
 TF_CALL_half(REGISTER_KERNEL);
 #undef REGISTER_KERNEL
 
+class LRNGradInitHelper : public InitializationHelper {
+ public:
+  using Attributes = LRNAttributes;
+
+  LRNGradInitHelper(OpKernelContext* context,
+                std::shared_ptr<const Attributes> attr)
+      : attr_(attr) {
+    const Tensor& in_grads = context->input(0);
+    const Tensor& in_image = context->input(1);
+    const Tensor& out_image = context->input(2);
+
+    OP_REQUIRES(context, in_grads.dims() == 4 && in_image.dims() == 4,
+                errors::InvalidArgument("inputs must be 4-dimensional"));
+    const int64 batch = in_grads.dim_size(0);
+    const int64 rows = in_grads.dim_size(1);
+    const int64 cols = in_grads.dim_size(2);
+    const int64 depth = in_grads.dim_size(3);
+    OP_REQUIRES(
+        context,
+        in_image.dim_size(0) == batch && in_image.dim_size(1) == rows &&
```

Comment:
Darn, this awkward wrapping really makes it unnecessarily difficult to read code. Since we're not (realistically) going to merge this with TF 1.5 master anyway, can we use a sane clang format for our subset 😅? I suppose since we're not all nicely partitioned off within a subfolder, like with ORT, we can't easily do so.